test/gurka/test_gtfs.cc

```diff
@@ -200,16 +200,18 @@ TEST(GtfsExample, WriteGtfs) {
   }
 
   // write agency.txt
-  struct Agency ttc {
```

Comment:
These changes are the most tedious changes.

I used GPT to rewrite these C++20 designated initialisers to compliant C++17. I've tried to go through line by line, but extra set of eyes would be good. (I guess any misses would manifest as failing tests)

Reply:
>  (I guess any misses would manifest as failing tests)

I did go through a bit, but yeah, I'm counting on this mostly. thanks for reverting this to c++17:)